internal/controllers/clusterextension_controller.go

```diff
@@ -112,82 +179,232 @@ func checkForUnexpectedFieldChange(a, b ocv1alpha1.ClusterExtension) bool {
 	return !equality.Semantic.DeepEqual(a, b)
 }
 
+func (r *ClusterExtensionReconciler) handleResolutionErrors(ext *ocv1alpha1.ClusterExtension, err error) (ctrl.Result, error) {
+	var aggErrs utilerrors.Aggregate
+	if errors.As(err, &aggErrs) {
+		for _, err := range aggErrs.Errors() {
+			errorMessage := err.Error()
+			if strings.Contains(errorMessage, "no package") {
```

Comment:
Relying on the error message (here and below) does not sound like a good idea to me.

Reply:
+1, if we need to perform functionality dependent on the error that occurred we should build custom error types that are returned by the method/function being called and passed into this function as a parameter

Reply:
See #878